src/libxtp/basisset.cc

```diff
@@ -23,136 +23,100 @@ namespace votca {
 namespace xtp {
 
 Index FindLmax(const std::string& type) {
-  Index lmax = std::numeric_limits<Index>::min();
-  // single type shells
-  if (type.length() == 1) {
-    if (type == "S") {
-      lmax = 0;
-    } else if (type == "P") {
-      lmax = 1;
-    } else if (type == "D") {
-      lmax = 2;
-    } else if (type == "F") {
-      lmax = 3;
-    } else if (type == "G") {
-      lmax = 4;
-    } else if (type == "H") {
-      lmax = 5;
-    } else if (type == "I") {
-      lmax = 6;
-    } else {
-      throw std::runtime_error("FindLmax: Shelltype not known");
-    }
+
+  assert(!type.empty() && "Shelltype most be non empty!");
+  const char t = type.back();
+  Index lmax = 0;
+  if (t == 'S') {
+    lmax = 0;
+  } else if (t == 'P') {
+    lmax = 1;
+  } else if (t == 'D') {
+    lmax = 2;
+  } else if (t == 'F') {
+    lmax = 3;
+  } else if (t == 'G') {
+    lmax = 4;
+  } else if (t == 'H') {
+    lmax = 5;
+  } else if (t == 'I') {
+    lmax = 6;
   } else {
-    for (Index i = 0; i < Index(type.length()); ++i) {
-      std::string local_shell = std::string(type, i, 1);
-      Index test = FindLmax(local_shell);
-      if (test > lmax) {
-        lmax = test;
-      }
-    }
```

Comment:
What will happen if there is a SP shell or similar in the basis?

Reply:
it always takes the last character of the shell. These routines now assume that a) the shell characters are sorted and continous.

Reply:
Makes sense.